api/types.go

```diff
@@ -12,21 +15,31 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
+	IssuerUrl    *string `json:"issuer_url"`
+	ClientId     *string `json:"client_id"`
+	ClientSecret *string `json:"client_secret"`
+	Audience     *string `json:"audience"`
 	Enabled      *bool
+	Filename     *string
 }
 
 type TokenVerificationConfig struct {
 	Enabled *bool
 	Url     *string
 }
 
-func (c *OIDCConfig) Validate() {
+func (c *OIDCConfig) ReadAndValidate() {
```

Comment:
The way this method is written is not testing friendly as it couples the logic to reading a file.
Getting a reader would be more testable friendly. We can tackle that in another PR if desired